### PR TITLE
Find definition for external declared entity

### LINKED ISSUE #706: Find definition for external declared entity
Like #625 but for external entities.

### PR DESCRIPTION
**Pull request by angelozerr**
Find definition for external declared entity

Fixes #706

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
To test this PR, create a `base.dtd` file like this:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<!ELEMENT root-element (#PCDATA)>
<!ENTITY external "A B">
<!ENTITY external2 "EXTERNALLY DECLARED ENTITY">
```

Create a `file.xml` file like this:

```xml
<?xml version="1.0" standalone="no"?>
<!DOCTYPE root-element SYSTEM "base.dtd" [
	<!ENTITY local "LOCALLY DECLARED ENTITY">
]>   
<root-element>
&local;       
&external;     
&undeclared;
</root-element>
```

Do a Ctrl+Click on &external, it should open base.dtd and select the proper entity name.

**Comment by angelozerr** [comment]
@fbricon @xorye I fixed test, please review this PR to close the entity hyperlink topic.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/uriresolver/CacheResourcesManagerTest.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ public void testUnavailableCache() throws Exception {
 	public void testAvailableCache() throws Exception {
 		FileServer server = new FileServer();
 		server.start();
-		String uri = server.getUri("/dtd/web-app_2_3.dtd");
+		String uri = server.getUri("/dtd/base.dtd");
```
Comment (fbricon) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/uriresolver/CacheResourcesManagerTest.java:92]: how is that related to this PR?

Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/uriresolver/CacheResourcesManagerTest.java]: Yes you are right, I removed this update.

**Comment by fbricon** [comment]
Nice work @angelozerr!
